app/models/early_career_payments/dqt_record.rb

```diff
@@ -36,16 +36,16 @@ def eligible_induction?
 
     # TODO: May need to prioritise subject chosen by highest award amount?
     def eligible_itt_subject_for_claim
-      return nil if !itt_subjects || itt_subjects.empty?
+      raise if !itt_subject_groups || itt_subject_groups.empty?
 
       year = itt_year || claim.eligibility.itt_academic_year # The user may have supplied this manually if it was missing from the DQT record
 
-      return nil unless year
+      raise unless year
```

Comment:
It's not great that we are are using an exception to control flow here, but acceptable. As an alternative, we could just use guard clauses with `return :none_of_the_above`.

Reply:
I've combined the guards so there is less duplication when removing the exceptions.

Reply:
I was curious so I had a look at `JourneySubjectEligibilityChecker`, something definitely for another time, but it could/should be refactored to raise custom exceptions that we can then rescue instead of catching all... otherwise that fallback could be a nightmare to debug one day. Looks better now anyway 👍

Reply:
Agreed!